examples/bri-3/src/bri/merkleTree/agents/merkleTreeStorage.agent.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@nestjs/common';
+import { InjectMapper } from '@automapper/nestjs';
+import { Mapper } from '@automapper/core';
+import { PrismaService } from '../../../../prisma/prisma.service';
+import MerkleTree from 'merkletreejs';
+import { BpiMerkleTree } from '../models/bpiMerkleTree';
+
+@Injectable()
+export class MerkleTreeStorageAgent extends PrismaService {
+  constructor(@InjectMapper() private mapper: Mapper) {
+    super();
+  }
+
+  async getMerkleTreeById(id: string): Promise<BpiMerkleTree> {
+    const merkleTreeModel = await this.bpiMerkleTree.findUnique({
+      where: { id },
+    });
+
+    if (!merkleTreeModel) {
+      return null;
+    }
+
+    const merkleTree = new BpiMerkleTree(
+      merkleTreeModel.id,
+      MerkleTree.unmarshalTree(merkleTreeModel.merkleTree),
+    );
+    return this.mapper.map(merkleTree, BpiMerkleTree, BpiMerkleTree);
+  }
+
+  async createNewMerkleTree(merkleTree: BpiMerkleTree): Promise<BpiMerkleTree> {
+    const treeIn = {
+      id: merkleTree.id,
+      merkleTree: MerkleTree.marshalTree(merkleTree.merkleTree),
+    };
+    const newMerkleTreeModel = await this.bpiMerkleTree.create({
+      data: {
+        ...treeIn,
+      },
+    });
+
+    const treeOut = {
+      id: newMerkleTreeModel.id,
+      merkleTree: MerkleTree.unmarshalTree(merkleTree.merkleTree),
```

Comment:
you probably meant to unmarshal either treeIn.merkleTree or newMerkleTreeModel.data.merkleTree?

Why do we need treeOut at all? Cant we just return method param merkleTree as it is the same as treeOut?

Reply:
I was just taking the ORM return and unmarshaling it from a string back to MerkleTree, but now just returning the param merkletree as suggested.